drivers/volume/portworx/portworx.go

```diff
@@ -1982,9 +1982,34 @@ func (p *portworx) DescribeSnapshot(snapshotData *crdv1.VolumeSnapshotData) (*[]
 	return &snapConditions, true, err
 }
 
-// TODO: Implement FindSnapshot
+// FindSnapshot return snapshotdata source for created portworx snapshot
+// Note: we wait for underlying driver snapshot creation in SnapshotCreate() only,
+// This will be called by snapshotter by -
+// Every time new snapshot is created
 func (p *portworx) FindSnapshot(tags *map[string]string) (*crdv1.VolumeSnapshotDataSource, *[]crdv1.VolumeSnapshotCondition, error) {
-	return nil, nil, &errors.ErrNotImplemented{}
+	if tags == nil || len(*tags) == 0 {
+		return nil, nil, fmt.Errorf("empty tags recevied for snapshots")
+	}
+	name := (*tags)[snapshotter.CloudSnapshotCreatedForVolumeSnapshotNameTag]
+	namespace := (*tags)[snapshotter.CloudSnapshotCreatedForVolumeSnapshotNamespaceTag]
```

Comment:
Do we need an 'ok'/nil check here?

Reply:
tags check handled above any of name/namespace empty we catch it below